Source/DafnyCore/Backends/Rust/Dafny-compiler-rust.dfy

```diff
@@ -3774,17 +3775,25 @@ module {:extern "DCOMP"} DafnyToRustCompiler {
       }
     }
 
-    static method EmitCallToMain(fullName: seq<Name>) returns (s: string) {
-      s := "\nfn main() {\n";
-      var i := 0;
-      while i < |fullName| {
-        if i > 0 {
-          s := s + "::";
-        }
-        s := s + escapeName(fullName[i]);
-        i := i + 1;
+    method EmitCallToMain(companion: Expression, mainMethodName: string, hasArgs: bool) returns (s: string) {
+      s := "\nfn main() {";
+      if hasArgs {
```

Comment:
This test of whether the method has arguments is there for historical reasons, but it seems that SinglePassCodeGenerator always add an implicit formal for arguments on main methods even if it's not declared, so I'm not sure the case `!hasArgs` is tested.